File: app/src/main/java/com/example/habithero/SoundHelper.java
package com.example.habithero;

import android.content.Context;
import android.media.MediaPlayer;

public class SoundHelper {

    // Enum to define different sound types
    public enum SoundType {
        COMPLETION, DELETE, CLICK, TAP, DENY, NEXT
    }

    // Method to play different sounds based on the type
    public static void playSound(Context context, SoundType soundType) {
        int soundResourceId = getSoundResourceId(soundType);
        MediaPlayer mediaPlayer = MediaPlayer.create(context, soundResourceId);
        mediaPlayer.setOnCompletionListener(MediaPlayer::release);
        mediaPlayer.start();
    }

    // Method to get the resource ID of the sound based on the type
    private static int getSoundResourceId(SoundType soundType) {
        switch (soundType) {
            case COMPLETION:
                return R.raw.completion_sound;
            case DELETE:
                return R.raw.delete_sound;
            case CLICK:
                return R.raw.click_sound;
            case TAP:
                return R.raw.tap_sound;
            case DENY:
                return R.raw.deny_sound;
            case NEXT:
                return R.raw.next_sound;
            default:
                throw new IllegalArgumentException("Unknown SoundType: " + soundType);
        }
    }
}

File: app/src/main/java/com/example/habithero/SoundHelper.kt
package com.example.habithero

import android.content.Context
import android.media.MediaPlayer

object SoundHelper {

    enum class SoundType {
        COMPLETION, DELETE, CLICK, TAP, DENY, NEXT
    }

    fun playSound(context: Context, soundType: SoundType) {
        val soundResourceId = getSoundResourceId(soundType)
        val mediaPlayer = MediaPlayer.create(context, soundResourceId)
        mediaPlayer.setOnCompletionListener { it.release() }
        mediaPlayer.start()
    }

    private fun getSoundResourceId(soundType: SoundType): Int {
        return when (soundType) {
            SoundType.COMPLETION -> R.raw.completion_sound
            SoundType.DELETE -> R.raw.delete_sound
            SoundType.CLICK -> R.raw.click_sound
            SoundType.TAP -> R.raw.tap_sound
            SoundType.DENY -> R.raw.deny_sound
            SoundType.NEXT -> R.raw.next_sound
        }
    }
}
